cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -1164,9 +1165,13 @@ private V1PodSpec createPodSpec(Type resourceType, Location runtimeConfigLocatio
         .filter(
             envVar -> !envVar.getName().equals(WorkloadIdentityUtil.WORKLOAD_IDENTITY_ENV_VAR_KEY))
         .collect(Collectors.toMap(V1EnvVar::getName, V1EnvVar::getValue));
-    // Add all environments of the the main runnable for the init container.
+    // Add all environments of the main runnable for the init container except GCE_METADATA_HOST.
```

Comment:
Can we add a comment explaining why we filter `GCE_METADATA_HOST` out?

Reply:
done.